Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/util/GenericDBUtils.java

```diff
@@ -168,6 +168,21 @@ public static void rollbackTransaction(Transaction transaction) {
         return entity;
     }
 
+    public static <T> T readBy(Session session, Class<T> clazz, String id) {
+        LOG.trace("GenericDBUtil.readBy");
+        T entity = null;
+        try {
+            entity = session.get(clazz, id);
+            LOG.trace("GenericDBUtil.readBy - entity = session.get()");
+        } catch (HibernateException | NullPointerException e) {
+            LOG.error("Exception during read caused by : {}", e.getMessage(), e);
```

Comment:
Either we log or rethrow exception but not both.  See this https://sonarcloud.io/organizations/default/rules?open=squid%3AS2139&rule_key=squid%3AS2139